R/utils.R

```diff
@@ -1,299 +1,529 @@
-
-################################################################################
-# internal methods loaded from other packages
-
-.get_mat_from_sce <- scater:::.get_mat_from_sce
-.get_mat_for_reddim <- scater:::.get_mat_for_reddim
-
-################################################################################
-# integration with other packages
-
-.require_package <- function(pkg){
-    if(!requireNamespace(pkg, quietly = TRUE)){
-    stop("'",pkg,"' package not found. Please install the '",pkg,"' package ",
-         "to use this function.", call. = FALSE)
-    }
-}
-
-################################################################################
-# testing
-
-.is_a_bool <- function(x){
-  is.logical(x) && length(x) == 1L && !is.na(x)
-}
-
-.is_non_empty_character <- function(x){
-  is.character(x) && all(nzchar(x))
-}
-
-.is_non_empty_string <- function(x){
-  .is_non_empty_character(x) && length(x) == 1L
-}
-
-.is_a_string <- function(x){
-  is.character(x) && length(x) == 1L
-}
-
-.is_an_integer <- function(x){
-    is.numeric(x) && length(x) == 1L && x%%1==0
-}
-
-.are_whole_numbers <- function(x){
-  tol <- 100 * .Machine$double.eps
-  abs(x - round(x)) <= tol && !is.infinite(x)
-}
-
-.is_numeric_string <- function(x){
-  x <- as.character(x)
-  suppressWarnings({x <- as.numeric(x)})
-  !is.na(x)
-}
-
-.is_function <- function(x){
-  typeof(x) == "closure" && is(x, "function")
-}
-
```

Comment:
Why it seems that all these functions have changed

Reply:
Oh its because to run GHA, I put commits against `mia:master`. So it includes all the earlier commits you pushed in `getter_and_setters`

Reply:
okay. I don't have access.